feature_creation_scripts/bedmachine_antarctica_v3_1000m_transect/bedmachine_antarctica_v3_1000m_transect.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python
+"""
+Download the BedMachine topography
+----------------------------------
+1. Go tohttps://nsidc.org/data/nsidc-0756/versions/3
+2. Click on "HTTPS File System"
+3. Log in or create and account.
+5. Under "1970.01.01" choose "BedMachineAntarctica-v3.nc"
+
+On Anvil or Chrysalis, BedMachine Antarctica v3 is available so create a
+symlink with:
+ln -s /lcrc/group/e3sm/public_html/mpas_standalonedata/mpas-ocean/bathymetry_database/BedMachineAntarctica-v3.nc
+"""
+
+import shutil
+
+import matplotlib.pyplot as plt
+import numpy as np
+import shapely
+import xarray as xr
+from pyproj import Transformer
+
+from geometric_features import GeometricFeatures, FeatureCollection
+from geometric_features.feature_collection import _round_coords
+from geometric_features.utils import write_feature_names_and_tags
+
+
+def get_longest_contour(contour_value, author):
+
+    def stereo_to_lon_lat(x, y):
+        lat, lon = transformer.transform(x, y)
+        return lon, lat
+
+    with xr.open_dataset('BedMachineAntarctica-v3.nc') as ds:
+        # the bed but only in the open ocean or under floating ice
+        bed = xr.where(np.logical_or(ds.mask == 0, ds.mask == 3),
+                       ds.bed, 0.).values
+        x = ds.x.values
+        y = ds.y.values
+
+    # plot contours
+    plt.figure()
+    cs = plt.contour(x, y, bed, (contour_value,))
+    paths = cs.allsegs[0]
+
+    path_lengths = [len(paths[i]) for i in range(len(paths))]
+    ilongest = np.argmax(path_lengths)
+
+    v = paths[ilongest]
+    x = v[:, 0]
+    y = v[:, 1]
+
+    # the starting index should be the point closest to the positive x axis
+    mask = x > 0.
+    indices = np.arange(y.shape[0])[mask]
+    index = np.argmin(np.abs(y[mask]))
+    first = indices[index]
+    x = np.append(x[first:], x[:first])
+    y = np.append(y[first:], y[:first])
+
+    # Antarctic stereographic to lat/lon
+    transformer = Transformer.from_crs('epsg:3031', 'epsg:4326')
+
+    transect = shapely.geometry.LineString([(i[0], i[1]) for i in zip(x, y)])
+
+    # cut a square out at around the first point at 90 degrees longitude to
+    # prevent the shape from being a closed loop
+    width = 30e3
+    x0 = x[0]
+    y0 = y[0]
+    square = shapely.geometry.Polygon([(x0 - width, y0 - width),
+                                       (x0 + width, y0 - width),
+                                       (x0 + width, y0 + width),
+                                       (x0 - width, y0 + width),
+                                       (x0 - width, y0 - width)])
+
+    difference = transect.difference(square)
```

Comment:
I ended up needing to cut a little square out of the transect.  The mask creator (specifically the code that finds edge signs) was totally confused by the transect being a closed loop.  We could make the square less than 60 km in size if we want to.

Here's what that gap ends up looking like:
![edge_sign](https://github.com/MPAS-Dev/geometric_features/assets/4179064/0d91b54d-c218-4937-adcf-9a84315a7f04)